### PR TITLE
free(NULL) no-op

### PR DESCRIPTION
**Pull request by gperciva**

### CODE REVIEW

File: libcperciva/crypto/crypto_aes.c [hunk 1]
```diff
@@ -151,6 +151,10 @@ void
 crypto_aes_key_free(struct crypto_aes_key * key)
 {
 
+	/* Behave consistently with free(NULL). */
```
Comment (cperciva) [inline libcperciva/crypto/crypto_aes.c]: This should be after the `#ifdef CPUSUPPORT_X86_AESNI` block.

File: tar/ccache/ccache_read.c [hunk 2]
```diff
@@ -162,6 +162,10 @@ callback_free(void * cookie, uint8_t * s, size_t slen, void * rec)
 {
 	struct ccache_record * ccr = rec;
 
+	/* Behave consistently with free(NULL). */
```
Comment (cperciva) [inline tar/ccache/ccache_read.c]: Not here; this isn't a nameofdatastructure_free function.

File: tar/chunks/chunks_directory.c [hunk 3]
```diff
@@ -82,6 +82,10 @@ callback_free(void * rec, void * cookie)
 {
 	struct chunkdata * ch = rec;
 
+	/* Behave consistently with free(NULL). */
```
Comment (cperciva) [inline tar/chunks/chunks_directory.c]: Not a _free function.

File: tar/storage/storage_read.c [hunk 4]
```diff
@@ -613,6 +613,10 @@ callback_cache_free(void * record, void * cookie)
 {
 	struct read_file_cached * CF = record;
 
+	/* Behave consistently with free(NULL). */
```
Comment (cperciva) [inline tar/storage/storage_read.c]: Not a _free function.

**Comment by gperciva** [comment]
Updated

**Pull request merged**